main/src/library/MutList.flix

```diff
@@ -68,12 +68,16 @@ mod MutList {
         MutList(rc, ref Array.init(rc, f, c) @ rc, ref d @ rc)
 
     ///
-    /// Optionally returns the element at position `i` in the mutable list `v`.
+    /// Optionally returns the element at position `n` in the mutable list `v`.
     ///
-    pub def nth(i: Int32, v: MutList[a, r]): Option[a] \ r =
+    pub def nth(n: Int32, v: MutList[a, r]): Option[a] \ r =
         let MutList(_, a, l) = v;
-        if (0 <= i and i < deref l)
-            Array.nth(i, deref a)
+        if (0 <= n and n < deref l)
```

Comment:
It seems like we are checking twice. Both here and now inside Array.nth. 

I also strikes me as wrong that `MutList.nth` should return an Option since we are just in the process of having all `nth` functions return the element directly + exception. Can we change it do that instead?

Reply:
MutList has to check, because the Array can be longer than the length of the MutList (the minimum capacity is 8).
Would Array.get would be acceptable here too?

Reply:
Lets do one check in MutList and then use Array.get internally.